### DISCUSSION TITLE
silent invalidateQueries

### DISCUSSION THREAD

**Discussion by islamBelabbes** [TanStack/query#7281] [Q&A]
hey,
I have some posts list that each has a ( like or dislike ) mutation, and I use queryClient.invalidateQueries to handle an optimistic update.

At the same time, I use a category list to filter based on category.

(change query key on category click )

now when I invalidate queries 
```ts
  if (isFetching)
    return <HashLoader />;
```

this result isFetching to be true each time I like or dislike, therefore, the loading will show.

the only idea that comes to mind now is doing setQueryData
but is there a way to exclude the loader when invalidateQueries? ( maybe doing a silence refresh )

**Comment by islamBelabbes** [comment]
edit : this issue only happens when I have initialdata
as isLoading will no longer be useful and I think its the only way I can do what I meant to do ?

**Reply by TkDodo** [reply]
if you only want a loader for the first load, check for `isLoading`. if you have initialData, you won't get a loader - that is on purpose. maybe this helps: https://tkdodo.eu/blog/react-query-fa-qs#why-is-the-queryfn-not-called
